src/gui/entry/EntryView.cpp

```diff
@@ -144,24 +268,262 @@ Entry* EntryView::entryFromIndex(const QModelIndex& index)
     }
 }
 
-void EntryView::switchToEntryListMode()
+/**
+ * @author Fonic <https://github.com/fonic>
+ * Switch to list mode, i.e. list entries of group (NOTE: previously named
+ * 'switchToGroupMode')
+ */
+void EntryView::switchToListMode()
 {
-    m_sortModel->hideColumn(0, false);
+    /* Check if already in this mode */
+    if (!m_inSearchMode) {
+        return;
+    }
 
-    m_sortModel->sort(1, Qt::AscendingOrder);
-    m_sortModel->sort(0, Qt::AscendingOrder);
-    sortByColumn(0, Qt::AscendingOrder);
+    /**
+     * Use header()->hideSection() instead of m_sortModel->hideColumn() as
+     * the latter messes up column indices, interfering with code relying on
+     * proper indices
+     */
+    header()->hideSection(EntryModel::ParentGroup);
 
-    m_inEntryListMode = true;
+    m_inSearchMode = false;
 }
 
-void EntryView::switchToGroupMode()
+/**
+ * @author Fonic <https://github.com/fonic>
+ * Switch to search mode, i.e. list search results (NOTE: previously named
+ * 'switchToEntryListMode')
+ */
+void EntryView::switchToSearchMode()
 {
-    m_sortModel->hideColumn(0, true);
+    /* Check if already in this mode */
+    if (m_inSearchMode) {
+        return;
+    }
+
+    /*
+     * Use header()->showSection() instead of m_sortModel->hideColumn() as
+     * the latter messes up column indices, interfering with code relying on
+     * proper indices
+     */
+    header()->showSection(EntryModel::ParentGroup);
 
-    m_sortModel->sort(-1, Qt::AscendingOrder);
-    m_sortModel->sort(0, Qt::AscendingOrder);
-    sortByColumn(0, Qt::AscendingOrder);
+    /*
+     * Always set sorting to column 'Group', as it does not feel right to have
+     * the last known sort configuration of search view restored by 'Database
+     * WidgetStateSync', which is what happens without this
+     */
+    m_sortModel->sort(EntryModel::ParentGroup, Qt::AscendingOrder);
+    sortByColumn(EntryModel::ParentGroup, Qt::AscendingOrder);
+
+    m_inSearchMode = true;
+}
+
+/**
+ * @author Fonic <https://github.com/fonic>
+ * Get current state of 'Hide Usernames' setting (NOTE: just pass-through for
+ * m_model)
+ */
+bool EntryView::hideUsernames() const
+{
+    return m_model->hideUsernames();
+}
+
+/**
+ * @author Fonic <https://github.com/fonic>
+ * Set state of 'Hide Usernames' setting (NOTE: just pass-through for m_model)
+ */
+void EntryView::setHideUsernames(const bool hide)
+{
+    m_model->setHideUsernames(hide);
+}
+
+/**
+ * @author Fonic <https://github.com/fonic>
+ * Get current state of 'Hide Passwords' setting (NOTE: just pass-through for
+ * m_model)
+ */
+bool EntryView::hidePasswords() const
```

Comment:
This function being const is the only thing showing that is it not actually hiding the password. I recommend renaming to isPasswordsHidden(). Rename the setter accordingly.

Reply:
Done